In [37]:
import Pkg; 
Pkg.add("PyCall")
Pkg.add("BenchmarkTools")

  Resolving package versions...
   Updating `C:\Users\Aliaga\.julia\environments\v1.4\Project.toml`
 [no changes]
   Updating `C:\Users\Aliaga\.julia\environments\v1.4\Manifest.toml`
 [no changes]
  Resolving package versions...
   Updating `C:\Users\Aliaga\.julia\environments\v1.4\Project.toml`
 [no changes]
   Updating `C:\Users\Aliaga\.julia\environments\v1.4\Manifest.toml`
 [no changes]


Probando la sucesión de Fibonacci con Julia

In [38]:
function fibonacciJulia(a) 
    t1 = 0 
    t2 = 1
    nextTerm = 0
    for i = 1:a
        if i == 1
            t1
            continue
        elseif i == 2
            t2
            continue
        end
        nextTerm = BigInt(t1) + BigInt(t2)
        t1 = BigInt(t2)
        t2 = BigInt(nextTerm)
        nextTerm
            
    end
end

fibonacciJulia (generic function with 1 method)

In [39]:
using BenchmarkTools
# Fibonnacci with Julia
@btime fibonacciJulia(5000)

  913.900 μs (10004 allocations: 1.24 MiB)


Julia 
   in my pc -> 898.299 μs (10004 allocations: 1.24 MiB)

## Fibonacci with Python

In [40]:
import PyCall

PyCall.py"""
#Fibonacci
# Complejidad O(n)

def fibonacciPython(n):
    t1 = 0
    t2 = 1
    nextTerm = 0

    for x in range(1,n):
        if x == 1:
            t1
        elif x == 2:
            t2

        nextTerm = t1 + t2
        t1 = t2
        t2 = nextTerm
"""

@btime PyCall.py"fibonacciPython"(5000)

  1.206 ms (9 allocations: 336 bytes)


Python 
in my pc -> 1.177 ms

#Summary

Mean Time:

Fibo with 5000 numbers sucesion

#### Python -> 1.177 ms
#### Julia  -> 898.299 μs = 0.89899 ms

